src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs

```diff
@@ -422,6 +422,45 @@ public void DestinationFileIsTruncatedWhenItsLargerThanSourceFile()
 
             Assert.Equal(content, File.ReadAllBytes(destPath));
         }
+
+        [Theory, MemberData(nameof(TestData.ValidFileNames), MemberType = typeof(TestData))]
+        public void CopyWithProblematicNames(string fileName)
+        {
+            DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
+            string sourcePath = Path.Combine(testDir.FullName, fileName);
+            string destPath = Path.Combine(testDir.FullName, fileName + "_copy");
+
+            File.Create(sourcePath).Dispose();
+            Copy(sourcePath, destPath);
+
+            Assert.True(File.Exists(sourcePath));
+            Assert.True(File.Exists(destPath));
+        }
+
+        [Theory, MemberData(nameof(TestData.WindowsTrailingProblematicFileNames), MemberType = typeof(TestData))]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void WindowsCopyWithTrailingSpacePeriod_ViaExtendedSyntax(string fileName)
+        {
+            // On Windows, files with trailing spaces/periods must use \\?\ prefix for creation
```

Comment:
Is there a .NET documentation at learning.microsoft.com where this is explained that we can link to from here instead of the verbose explanation? If not, we should create one - this is important detail that should be documented in API docs.